### ISSUE TITLE
"docker is starting" but docker command no response   

### ISSUE THREAD

**Issue by EKWebMaster** [docker/for-mac#3024]
ID: EED01C1A-30BB-44A3-9009-6970325C1B51

  - [x ] I have tried with the latest version of my channel (Stable or Edge)
  - [ x] I have uploaded Diagnostics
  - Diagnostics ID:
EED01C1A-30BB-44A3-9009-6970325C1B51

### Expected behavior
after start the docker, it needs to be available to use
### Actual behavior
"docker is starting" but not actually started and all docker command not working without response
### Diagnostic logs

```
Docker for Mac: version: 18.03.1-ce-mac65 (dd2831d4b7421cf559a0881cc7a5fdebeb8c2b98)
macOS: version 10.12 (build: 16A323)
logs: /tmp/EED01C1A-30BB-44A3-9009-6970325C1B51/20180621-195357.tar.gz
failure: docker ps failed: (Failure "docker ps: timeout after 10.00s")
[OK]     vpnkit
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         docker ps failed
[OK]     disk

**Comment by RonSidell** [comment]
Even though Docker was working earlier today, I get these same exact symptoms now.  The last thing I did before it failed to restart is that I altered the entries in the sharing section of the preferences.

I ran diagnostics, resulting in this diagnostic id: 70CFF7E5-146E-48C9-B7F1-5D1869443A71
and these results:
Docker for Mac: version: 18.03.1-ce-mac65 (dd2831d4b7421cf559a0881cc7a5fdebeb8c2b98)
macOS: version 10.13.5 (build: 17F77)
logs: /tmp/70CFF7E5-146E-48C9-B7F1-5D1869443A71/20180621-194129.tar.gz
failure: docker ps failed: (Failure "docker ps: timeout after 10.00s")
[OK]     vpnkit
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         docker ps failed
[OK]     disk

**Comment by RonSidell** [comment]
Before switching to edge, I tried this from a terminal window:
screen ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/tty

It had no immediate effect but I switched to other tasks on my Mac and eventually Docker went from "starting" to "running."  That was reproducible, but the time required for Docker to enter the running state seemed random and could take up to 10 minutes.

I installed Edge:
$ docker version
Client:
 Version:      18.05.0-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   f150324
 Built:        Wed May  9 22:12:05 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.05.0-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.10.1
  Git commit:   f150324
  Built:        Wed May  9 22:20:16 2018
  OS/Arch:      linux/amd64
  Experimental: true

It starts up right away.  Edge was able to tell me that the folders I tried to add overlapped the folders that were already exported.  I'm a Docker nube, and the only reason I was adding shares was because of errors generated when I tried adding a container for gitlab, following their directions.  But following those instructions generated an error indicating that the container could not write to any  existing folder on any of my mounted drives.  The production version did not already have the bind mounts included with Edge.  These mounts: /Users, /Volumes, /private, /tmp are already bind mounted in Edge, but of those mounts, /tmp is the only folder on the bind mount list in the production version, and in my case, that seems related to the sudden failure of production Docker's starting correctly.
I'll continue using Edge for now.
Thanks for suggesting a workable solution.
